src/main/java/ee/tuleva/onboarding/config/DigiDocConfiguration.java

```diff
@@ -0,0 +1,22 @@
+package ee.tuleva.onboarding.config;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+
+@Configuration
+public class DigiDocConfiguration {
+
+    @Bean
+    @Profile("production")
+    public org.digidoc4j.Configuration digiDocConfigProd() {
+        return new org.digidoc4j.Configuration(org.digidoc4j.Configuration.Mode.PROD);
+    }
+
+    @Bean
+    @Profile("!production")
```

Comment:
If you have this kind of conditional beans then it is usually better for the negative case to use `@ConditionalOnMissingBean(Configuration.class)` instead of `@Profile~ makes the configuration a bit more resilient. Random example: current case would break if you add another Profile on top of the other bean and there would be two beans in the application context.

Even better would be configuring it in properties `@ConditionalOnProperty(prefix="digidoc.config", name="type", havingValue="prod")` then you can turn it on and off without switching the entire profile.